book/src/linking_and_dividing.md

```diff
@@ -0,0 +1,337 @@
+containing the static native method.
+
+```rust
+// Imports for `JClass and JEnv`
+use jni::objects::JClass;
+use jni::JNIEnv;
+```
+
+The next arguments map 1:1 with parameters specified in the Java method's
+signature. For example, if the signature was `void add(int a, int b)`, then the
+3rd and 4th arguments in the native function would be of type `jni::sys::jint`.
+For reference, these are the jni-rs types that you can include in your JNI
+signatures:
+
+```
+// from jni-rs
+pub type jint = i32;
+pub type jlong = i64;
+pub type jbyte = i8;
+pub type jboolean = u8;
+pub type jchar = u16;
+pub type jshort = i16;
+pub type jfloat = f32;
+pub type jdouble = f64;
+pub type jsize = jint;
+
+pub enum _jobject {}
+pub type jobject = *mut _jobject;
+pub type jclass = jobject;
+pub type jthrowable = jobject;
+pub type jstring = jobject;
+pub type jarray = jobject;
+pub type jbooleanArray = jarray;
+pub type jbyteArray = jarray;
+pub type jcharArray = jarray;
+pub type jshortArray = jarray;
+pub type jintArray = jarray;
+pub type jlongArray = jarray;
+pub type jfloatArray = jarray;
+pub type jdoubleArray = jarray;
+pub type jobjectArray = jarray;
+pub type jweak = jobject;
+```
+
+You'll notice that all of these are type aliases for existing signatures. In
+reality, JNI only passes down `jobject`, `jarray`, and various primitive types.
+Whenever possible, you should use the `jni-rs` type aliases above to take advantage of
+stronger typing and methods in the `jni-rs` project.
+
+## LookingGlass - Parameters and Return Types
+
+Since `test_call` in `LookingGlass` is static and takes no additional arguments,
+the function should be written as:
+
+```rust
+use jni::objects::JClass;
+use jni::JNIEnv;
+
+pub fn Java_com_github_imaginarypackagename_LookingGlass_test_1call(
+    _env: JNIEnv,
+    _class: JClass
+) {}
+```
+
+Although no arguments are used in this function, the first two arguments must
+always be in the native method's signature. The return type is void, and so
+nothing needs to be added to the Rust signature.
+
+Hypothetically, if the method returned an integer, it would look something like
+this:
+
+```rust
+use jni::objects::JClass;
+use jni::sys::jint;
+use jni::JNIEnv;
+
+pub fn Java_com_github_imaginarypackagename_LookingGlass_test_1call_1with_1jint(
+    _env: JNIEnv,
+    _class: JClass
+) -> jint {
+    123 as jint
+}
+```
+
+By the way, `LookingGlass` is almost done. Specifying ABIs and Disabling
+Mangling is last, and much less work.
+
+## Specifying the ABI and Disabling Mangling
+ABIs (Application Binary Interfaces) standardize low-level details, so that
+object code built using different compilers may still rely upon one another. In
+Rust, functions default to using the "Rust" ABI, which is usually desirable, but
+not for shared libraries.
+
+Therefore, it's necessary to explicitly set the ABI using `pub extern "system"`.
+For further information, see https://doc.rust-lang.org/std/keyword.extern.html
```

Comment:
My mdbook renders them as plain text :confused:

Reply:
Ah, got it. I think I fixed similar problems through the rest of the tutorial.